axelrod/strategies/lookerup.py

```diff
@@ -0,0 +1,135 @@
+from axelrod import Player, Actions
+import itertools
+
+C, D = Actions.C, Actions.D
+
+class LookerUp(Player):     
+
+    """     
+A strategy that uses a lookup table to
+decide what to do based on a combination of the last m rounds and the opponent's
+opening n moves. If there isn't enough history to do this (i.e. for the first m
+rounds) then cooperate.
+
+    The lookup table is implemented as a dict. The keys are 3-tuples giving the
+    opponents first n moves, self's last m moves, and opponents last m moves,
+    all as strings. The values are the moves to play on this round.  For
+    example, if
+
+        - the opponent started by playing C  
+        - my last move was a C the opponents
+        - last move was a D
+
+    the corresponding key would be
+
+        ('C', 'C', 'D')
+
+    and the value would contain the move to play on this turn.
+
+    Some well-known strategies can be expressed as special cases; for example
+    Cooperator is given by the dict:
+
+        {('', '', '') : 'C'}
+
+    where m and n are both zero. Tit For Tat is given by:
+
+    { 
+        ('', 'C', 'D') : 'D',         
+        ('', 'D', 'D') : 'D',         
+        ('', 'C', 'C') : 'C',
+        ('', 'D', 'C') : 'C',     
```

Comment:
I don't see it?

I'm basically saying it would be good to see an example where `m` or `n` are not <= 1.